iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -1154,7 +1154,17 @@ private async Task<Message> SendTwinRequestAsync(Message request, string rid, Ca
                         {
                             if (status >= 300)
                             {
-                                throw new IotHubException($"Request {rid} returned status {status}", isTransient: false);
+                                // The Hub team is refactoring the retriable status codes without breaking changes to the existing ones.
+                                // It can be expected that we may bring more retriable codes here in the future.
+                                // Retry for Http status code 429 (too many requests)
+                                if (status == 429)
+                                {
+                                    throw new IotHubThrottledException($"Request {rid} was throttled by the server");
```

Comment:
To avoid breaking change that can result from user searching for the text, how about 'adding' text instead of changing it?'

Also, by not setting isTransicent to false, does that make this fatal error instead?

_throw new IotHubException($"Request {rid} returned status {status} - throttled by server", isTransient: false);_

Reply:
Users shouldn't be parsing exception messages and branching based on that. We don't that part of the code contract.

`isTransient` should be true, because throttling is a temporary condition.